apps/studio/data/database-extensions/database-extensions-query.ts

```diff
@@ -0,0 +1,68 @@
+import { UseQueryOptions, useQuery, useQueryClient } from '@tanstack/react-query'
+import { get } from 'data/fetchers'
+import { useCallback } from 'react'
+import { ResponseError } from 'types'
+import { databaseExtensionsKeys } from './keys'
+
+export type DatabaseExtensionsVariables = {
+  projectRef?: string
+  connectionString?: string
+}
+
+export async function getDatabaseExtensions(
+  { projectRef, connectionString }: DatabaseExtensionsVariables,
+  signal?: AbortSignal
+) {
+  if (!projectRef) throw new Error('projectRef is required')
+
+  let headers = new Headers()
+  if (connectionString) headers.set('x-connection-encrypted', connectionString)
+
+  const { data, error } = await get('/platform/pg-meta/{ref}/extensions', {
+    params: {
+      header: {
+        'x-connection-encrypted': connectionString!,
+      },
+      path: {
+        ref: projectRef,
+      },
+    },
+    headers: Object.fromEntries(headers),
```

Comment:
`Object.fromEntries` shouldn't be needed here right? Can just do:
```suggestion
    headers,
```